### PR TITLE
Add Dango40

### PR DESCRIPTION
**Pull request by swiftrax**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/onekeyco/dango40/readme.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Dango40
+
+40% keyboard
+
+* Keyboard Maintainer: Swiftrax
+* Hardware Supported: Dango40
+* Hardware Availability: Dango40 PCB
+
+Make example for flashing this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/onekeyco/dango40/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/onekeyco/dango40/rules.mk [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/onekeyco/dango40/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/onekeyco/dango40/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/onekeyco/dango40/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/onekeyco/dango40/info.json [hunk 4]
```diff
@@ -0,0 +1,265 @@
+{
+   "keyboard_name":"Dango40",
+   "url":"https://github.com/swiftrax",
+   "maintainer":"Swiftrax",
+   "width":13.75,
+   "height":4,
```
Comment (drashna) [inline keyboards/onekeyco/dango40/info.json:6]:
```suggestion
```

File: keyboards/onekeyco/dango40/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    _______,  _______,  _______,            _______,            _______,  _______,            _______,  _______,  _______
+ )
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise)
+      tap_code(KC_VOLU);
+    else 
+      tap_code(KC_VOLD);
+}
```
Comment (drashna) [inline keyboards/onekeyco/dango40/keymaps/default/keymap.c:45]:
```suggestion
```

File: keyboards/onekeyco/dango40/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    _______,  _______,  _______,            _______,            _______,  _______,            _______,  _______,  _______
+ )
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise)
+      tap_code(KC_VOLU);
+    else 
+      tap_code(KC_VOLD);
+}
```
Comment (drashna) [inline keyboards/onekeyco/dango40/keymaps/via/keymap.c:57]:
```suggestion
```

File: keyboards/onekeyco/dango40/dango40.c [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "dango40.h"
```
Comment (drashna) [inline keyboards/onekeyco/dango40/dango40.c:17]:
```suggestion
#include "dango40.h"

bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (clockwise) {
        tap_code_delay(KC_VOLU, 10);
    } else {
        tap_code_delay(KC_VOLD, 10);
    }
    return true;
} 
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**